src/plugins/intel_npu/src/backend/src/zero_infer_request.cpp

```diff
@@ -399,6 +414,166 @@ void ZeroInferRequest::set_tensor(const ov::Output<const ov::Node>& port, const
+        auto batch = input.get_partial_shape()[batch_idx];
+
+        OPENVINO_ASSERT(batch.is_dynamic() || batch.get_length() == tensors_size,
+                        "set_input_tensors/set_tensors error. Input shape ",
+                        input.get_partial_shape(),
+                        "batch ",
+                        batch,
+                        "doesn't match with total blobs count: ",
+                        tensors_size);
+    }
+
+    auto batched_shape = tensors[BATCH_AXIS]->get_shape();
+    auto element_type = tensors[BATCH_AXIS]->get_element_type();
+    batched_shape[batch_idx] = tensors_size;
+    for (const auto& item : tensors) {
+        OPENVINO_ASSERT(item, "Unintialized tensor is provided!");
+        auto item_shape = item->get_shape();
+        item_shape[batch_idx] = batched_shape[batch_idx];
+        OPENVINO_ASSERT(item_shape == batched_shape && item->get_element_type() == element_type &&
+                            "set_input_tensors/set_tensors error. Tensor with element type ",
+                        item->get_element_type(),
+                        " and shape ",
+                        item_shape,
+                        " is not compatible with batched tensor with element type ",
+                        element_type,
+                        " and shape ",
+                        batched_shape);
+        OPENVINO_ASSERT(item->is_continuous(), "Strides for batched tensors should be default.");
+    }
+}
+
+void ZeroInferRequest::set_tensors(const ov::Output<const ov::Node>& port,
+                                   const std::vector<ov::SoPtr<ov::ITensor>>& tensors) {
+    OV_ITT_SCOPED_TASK(itt::domains::LevelZeroBackend, "set_tensors");
+    if (tensors.size() == 1) {
+        set_tensor(port, tensors[0]);
+        return;
+    }
+
+    auto foundPort = find_port(port);
+    OPENVINO_ASSERT(foundPort.found(), "Cannot find input tensor for port ", port);
+    if (!foundPort.is_input()) {
+        OPENVINO_THROW("set_input_tensors/set_tensors is not supported for output port.");
+    }
+
+    check_batched_tensors(port, tensors);
+
+    _userBatchedTensors[foundPort.idx] = tensors;
+
+    if (_initStructs->getMutableCommandListVersion()) {
+        if (_batchSize.has_value()) {
+            for (size_t i = 0; i < tensors.size(); i++) {
+                auto remoteTensor = std::dynamic_pointer_cast<ZeroRemoteTensor>(tensors[i]._ptr);
+
+                _levelZeroBatchedTensors[foundPort.idx].resize(tensors.size());
+                _batchedTensorsData[foundPort.idx].resize(tensors.size());
+
+                if (remoteTensor == nullptr) {
+                    bool tensorHasSameL0Context = false;
+
+                    ze_memory_allocation_properties_t desc = {};
+                    desc.stype = ZE_STRUCTURE_TYPE_MEMORY_ALLOCATION_PROPERTIES;
+                    auto res = zeMemGetAllocProperties(_initStructs->getContext(), tensors[i]->data(), &desc, nullptr);
```

Comment:
Should we create a private method for this check?

Reply:
done